pyartcd/pyartcd/pipelines/promote.py

```diff
@@ -370,6 +374,28 @@ def _get_image_stream_name(assembly_type: assembly.AssemblyTypes, arch: str):
         go_arch_suffix = go_suffix_for_arch(arch)
         return f'4-dev-preview{go_arch_suffix}' if assembly_type == assembly.AssemblyTypes.PREVIEW else f'release{go_arch_suffix}'
 
+    async def sync_rhcos_srpms(self, assembly_type, data):
+        # Sync potential pre-release source on which RHCOS depends. See ART-6419 for details.
+        major, minor = util.isolate_major_minor_in_group(self.group)
+        if (major > 4 or (major == 4 and minor >= 13)) and assembly_type in [assembly.AssemblyTypes.CANDIDATE, assembly.AssemblyTypes.STANDARD]:
+            src_output_dir = f"{self._working_dir}/rhcos_src_staging"
+            for arch in data['content']:
+                if arch != "multi":
+                    cmd = [
+                        "doozer",
+                        "--group", self.group,
+                        "--assembly", self.assembly,
+                        "config:rhcos-srpms",
+                        "--version", data["content"][arch]['rhcos_version'],
+                        "--arch", arch,
+                        "-o", src_output_dir,
+                    ]
+                    await exectools.cmd_assert_async(cmd, env=self._doozer_env_vars)
+            # Publish the clients to our S3 bucket.
+            await sync_dir_to_s3_mirror(src_output_dir, "/pub/openshift-v4/sources/packages/", "", "", False, False)
```

Comment:
For better readability, could you use keyword arguments instead of magical constants `False, False`?